Src/ILGPU/IR/Types/IRTypeContext.cs

```diff
@@ -579,6 +580,37 @@ public override void ClearCache(ClearCacheMode mode)
             PopulateTypeMapping();
         }
 
+        /// <summary>
+        /// Acquires a lock on the current <see cref="IRTypeContext"/> and
+        /// retrieves an <see cref="IEnumerable{TypeNode}"/> containing the
+        /// full type unification therewithin. 
+        /// </summary>
+        /// <returns>
+        /// The retrieved type unification as a generic collection.
+        /// </returns>
+        public IEnumerable<TypeNode> GetUnifiedTypeCollection()
+        {
+            using var readScope = typeLock.EnterReadScope();
+            foreach (var (type, _) in unifiedTypes)
+            {
+                yield return type;
+            }
+        }
+
+        /// <summary>
+        /// Serializes this instance using the given <see cref="IIRWriter"/>.
+        /// </summary>
+        /// <param name="writer">
+        /// The writer to use for serialization.
+        /// </param>
+        public void Write(IIRWriter writer)
+        {
+            foreach (var type in GetUnifiedTypeCollection())
+            {
```

Comment:
Nit: please remove non-required braces to match the overall ILGPU style.

Reply:
This is something I'm always hesitant to do when writing for OSS because people like their specific flavors of C syntax styling. Most commonly the nit is the opposite, hehe. Will fix